### PR TITLE
Fixing Uncertainty Principle

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
In this PR we're adding nullability specifiers to **ContextManager** and **LocalCoreDataService**. 

As a result, we're fixing the uncertainty principle: ~~the cat will be either dead or alive, but not both~~ we're nuking force unwraps // optional chaining, all over the codebase.

### To test:
- Please, verify that the unit tests work smoothly
- And please, review the changes. 

**Side note:**
To minimize the footprint of this PR, i've added a few force unwraps in snippets that look as follows:

```
ContextManager.shared.save(managedObject.managedObjectContext!)
```

Needs review: @aerych / @kurzee / @astralbodies 
Thanks in advance gentelmen!

### CODE REVIEW

**Comment by koke** [comment]
I think I might cry a little 😂

**Comment by kurzee** [comment]
@jleandroperez nice!

**Comment by jleandroperez** [comment]
@koke suggestions are always more than welcome!. I got really bugged by the `service?` statement we had all over!

@kurzee thanks sir!!

**Comment by kurzee** [comment]
@jleandroperez everything looks good on my first take, definitely could use some second eyes to be sure. I'll give it another take tomorrow as well.

File: WordPress/WordPressTest/TestContextManager.h [hunk 1]
```diff
@@ -9,13 +12,13 @@
...
 @interface TestContextManager : ContextManager
 
-@property (nonatomic, readwrite, strong) NSManagedObjectContext         *mainContext;
-@property (nonatomic, readwrite, strong) NSManagedObjectModel           *managedObjectModel;
-@property (nonatomic, readwrite, strong) NSPersistentStoreCoordinator   *persistentStoreCoordinator;
-@property (nonatomic, readonly,  strong) NSPersistentStoreCoordinator   *standardPSC;
-@property (nonatomic, readwrite, strong) XCTestExpectation              *testExpectation;
-@property (nonatomic, readwrite, assign) BOOL                           requiresTestExpectation;
-@property (nonatomic, readonly,  strong) NSURL                          *storeURL;
+@property (nonatomic, readwrite, strong)            NSManagedObjectContext         *mainContext;
```
Comment (astralbodies) [inline WordPress/WordPressTest/TestContextManager.h]: Sweet lord, why?

I'll allow it 😄

**Comment by jleandroperez** [comment]
Thanks everyone for the review(s)!!!

**Pull request merged**